### PR TITLE
Try to update FlighGear flatpak to 2020.1.3

### PR DESCRIPTION
**Pull request by Elinvention**
It seems to work, but I'm not sure because it's the first time that I try to
update a flatpak package.

### CODE REVIEW

**Comment by TingPing** [comment]
bot, build

**Comment by isantop** [comment]
I get this error on the command line when I try to run this:

```
flatpak run org.flightgear.FlightGear 
fgfs: /usr/lib/x86_64-linux-gnu/libQt5Core.so.5: version `Qt_5.14' not found (required by fgfs)
```

Looks like something may not be building correctly.

**Comment by AsciiWolf** [comment]
@isantop Maybe it needs a 5.14 runtime instead of 5.13?

**Comment by Elinvention** [comment]
Works™ on my Machine.

**Comment by Elinvention** [comment]
Could you please test, review and merge this? This should fix #14 and #15.

File: simgear-boost-1.73.patch
Comment (Vanadiae) [inline simgear-boost-1.73.patch:8]: well, this patch isn't required anymore, I checked in the simgear repo and there is already this line in it.

**Comment by Elinvention** [comment]
@Vanadiae thanks for your suggestions. Hopefully it should build now (even though it did build last time without ffmpeg).

File: flightgear-boost-1.73.patch
Comment (Vanadiae) [inline flightgear-boost-1.73.patch:8]: those flags are also already present upstream so the patch can be removed.

File: org.flightgear.FlightGear.yaml
Comment (Vanadiae) [inline org.flightgear.FlightGear.yaml:118]: so those two lines aren't required anymore.

File: org.flightgear.FlightGear.yaml
Comment (Vanadiae) [inline org.flightgear.FlightGear.yaml:75]: so those two lines aren't required anymore.

**Review by Vanadiae — CHANGES_REQUESTED** [review]

File: org.flightgear.FlightGear.appdata.xml
Comment (Vanadiae) [inline org.flightgear.FlightGear.appdata.xml:57]: this list should be ordered chronologically (newest first).

File: simgear-boost-1.73.patch
Comment (Elinvention) [reply simgear-boost-1.73.patch:8]: I guess we need to update to 2020.1.3 first.

Comment (Vanadiae) [reply simgear-boost-1.73.patch:8]: Oh if it's for the next version you can either keep them or update to 2020.1.3, at your convenience.

**Comment by Vanadiae** [comment]
@bilelmoussaoui (or an other flathub maintainer?), have you any idea of why the build is failing on some missing functions with ALSA ? A problem in the flatpak manifest maybe ? @Elinvention now I don't think it actually needs ffmpeg anymore…

**Comment by Vanadiae** [comment]
bot, build

**Comment by Vanadiae** [comment]
it seems FlightGear requires GLUT, since I found `GLUT NOT found, can't build FGPanel` in the build logs. Can you add the FreeGLUT from https://sourceforge.net/projects/freeglut/files/freeglut/3.2.1/ in the manifest ?

**Comment by Vanadiae** [comment]
> it seems FlightGear requires GLUT, since I found `GLUT NOT found, can't build FGPanel` in the build logs. Can you add the FreeGLUT from https://sourceforge.net/projects/freeglut/files/freeglut/3.2.1/ in the manifest ?

Well, you probably won't need as !13 will add it, so you'll just need to wait until it is merged then rebase.

**Comment by Elinvention** [comment]
`!13`?

**Comment by Vanadiae** [comment]
> `!13`?

I thought Github would link the merge request like gitlab does, but apparently not. I was talking about [merge request 13](https://github.com/flathub/org.flightgear.FlightGear/pull/13).

File: org.flightgear.FlightGear.yaml
Comment (Vanadiae) [inline org.flightgear.FlightGear.yaml:110]:
```suggestion
    url: https://iweb.dl.sourceforge.net/project/flightgear/release-2020.1/flightgear-2020.1.3.tar.bz2
```

The sha256 will need to be updated too.

File: org.flightgear.FlightGear.yaml
Comment (Vanadiae) [inline org.flightgear.FlightGear.yaml:98]: I don't think flightgear depends on gudev (as it's some GTK/GLib stuff but flightgear is c++), only libudev. It was removed from the shared modules as it's included in the FDO runtime, from which the KDE runtime inherits. So you can remove this line.

**Review by Vanadiae — CHANGES_REQUESTED** [review]
The other MR got merged. Can you please rebase your branch ? You'll have some conflicts to resolve though.  If you want I can continue the work on this MR, you just need to give me "collaborator" access to your repository in the settings of your repo's page on github.

**Comment by Vanadiae** [comment]
Thanks for the access to your branch! I'll give a go at rebasing your branch and check there isn't anything left to do then I'll push back.

**Comment by AsciiWolf** [comment]
It looks like the linker is not able to find the _asound_ library.

**Comment by Vanadiae** [comment]
> It looks like the linker is not able to find the _asound_ library.

Yes, ALSA is required. libasound is in the base FDO sdk, but only as shared library. libflite is *only* available as static library, from the KDE sdk (https://invent.kde.org/packaging/flatpak-kde-runtime/-/blob/qt5.14/org.kde.Sdk.json.in#L456). So the linker expects libasound to be statically linked, like libflite… From libflite's autoconf file:

```autoconf
AC_ARG_ENABLE( shared,  <== Here it declares the "shared" option
               [  --disable-shared     without shared library support],
	       [case "${enableval}" in
                 yes) shared=true ;;
                 no) shared=false ;;
                 *) AC_MSG_ERROR(bad value ${enableval} for shared options) ;;
                 esac],
               [shared=false]) <== but the default is to not build the shared library. And the KDE sdk doesn't pass --enable-shared when building.
```

So I think the solution is to get the kde sdk to use `--enable-shared` (or build libflite twice here). I don't think I have much choice but create a MR over at KDE sdk…

**Comment by Vanadiae** [comment]
I'll just include flite as part of the app, as /app has precedence over the runtime-provided libflite, that leaves in /usr. 🤞️

**Comment by Vanadiae** [comment]
Well well well, I've tested it and it is a great upgrade from the previous version! libflite seems to be working, as I could heard the annoying voice speaking. I still want to test this more before giving green light, but it looks overall good! Good news :)

**Review by Vanadiae — APPROVED** [review]

**Comment by Vanadiae** [comment]
I've tested it again, I didn't saw any problem at all (that weren't already there previously) so all good (except some things with wayland but I'll do a follow-up MR for this). Since I now have access to this repository, I'm gonna merge this. Thanks a lot @Elinvention for your MR! :)

**Comment by Elinvention** [comment]
@Vanadiae thanks for fixing these build issues and merging this!

**Pull request merged**
